raft/src/test/java/org/apache/kafka/raft/MockLog.java

```diff
@@ -42,10 +45,11 @@
     private final List<EpochStartOffset> epochStartOffsets = new ArrayList<>();
     private final List<LogBatch> log = new ArrayList<>();
     private LogOffsetMetadata highWatermark = new LogOffsetMetadata(0L);
+    private final Map<Long, UUID> offsetMetadataMap = new HashMap<>();
```

Comment:
I think another way to do this would be to have a field for `nextUUID` which gets updated after every append. Then we could remove this collection and store the respective UUID in the `LogEntry`.

Reply:
I've tried it first, but the issue is that besides the `fetch-info` the `endOffset()` call may also return this metadata that should be cached, but the end-offset is actually the one that is not appended yet -- i.e. we need to reserve the corresponding UUID for the next offset as well. So in the end I decided to go with just this simpler solution.

Reply:
Yeah, that's why I suggested storing `nextUUID` as a field. It would also be the UUID of the next entry to append to the log, so it would be valid to return it from `endOffset()`.

Reply:
Got it, okay I can do that --- in `updateHW` though we then need to do a search in the log batches list to find the corresponding offset's entry.